### ISSUE TITLE
error cloning virtual machine: Unable to access the virtual machine configuration: Unable to access file

### ISSUE THREAD

**Issue by angelichorsey** [terraform-providers/terraform-provider-vsphere#430]
### Terraform Version
```
$ terraform -v
Terraform v0.11.2
+ provider.vsphere v1.3.3
```
### vSphere Provider Version
```
.terraform
└── plugins
    └── linux_amd64
        ├── lock.json
        └── terraform-provider-vsphere_v1.3.3_x4
```
### Affected Resource(s)
- `vsphere_virtual_machine`
### Terraform Configuration Files

```hcl
data "vsphere_datacenter" "dc" {
  name = "#######"
}

data "vsphere_datastore" "datastore" {
  name          = "#######"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

data "vsphere_resource_pool" "pool" {
  name          = "#######"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

data "vsphere_network" "network" {
  name          = "#######"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

data "vsphere_virtual_machine" "template" {
  name          = "#######"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

provider "vsphere" {
  user                 = "${var.vsphere_user}"
  password             = "${var.vsphere_password}"
  vsphere_server       = "${var.vsphere_server}"
  allow_unverified_ssl = true
}

resource "vsphere_virtual_machine" "##########" {
  name             = "##########"
  resource_pool_id = "${data.vsphere_resource_pool.pool.id}"
  datastore_id     = "${data.vsphere_datastore.datastore.id}"
  folder           = "#######"

  num_cpus = 4
  memory   = 4096
  guest_id = "${data.vsphere_virtual_machine.template.guest_id}"

  scsi_type = "${data.vsphere_virtual_machine.template.scsi_type}"

  network_interface {
    network_id   = "${data.vsphere_network.network.id}"
    adapter_type = "${data.vsphere_virtual_machine.template.network_interface_types[0]}"
  }

  disk {
    label            = "disk0"
    size             = "${data.vsphere_virtual_machine.template.disks.0.size}"
    eagerly_scrub    = "${data.vsphere_virtual_machine.template.disks.0.eagerly_scrub}"
    thin_provisioned = "${data.vsphere_virtual_machine.template.disks.0.thin_provisioned}"
  }

  clone {
    template_uuid = "${data.vsphere_virtual_machine.template.id}"

    customize {
      linux_options {
        host_name = "##########"
        domain    = "#######"
      }

      network_interface {
        ipv4_address = "10.10.10.10"
        ipv4_netmask = 27
      }

      ipv4_gateway = "10.10.10.10"

      dns_server_list = [
        "10.10.10.10",
        "10.10.10.10",
        "10.10.10.10",
        "10.10.10.10",
        "10.10.10.10",
        "10.10.10.10",
      ]
    }
  }
}

```

### Debug Output
https://gist.github.com/angelichorsey/bae2f445814abe8a7259bcc576ce136c#file-tf_log_debug
### Expected Behavior
The resource should have been cloned from the template and provisioned into vsphere.
### Actual Behavior
The following error appeared:
```
Error: Error applying plan:

1 error(s) occurred:

* vsphere_virtual_machine.###########: 1 error(s) occurred:

* vsphere_virtual_machine.###########: error cloning virtual machine: Unable to access the virtual machine configuration: Unable to access file [##########]
```
### Steps to Reproduce
1. `terraform apply`
### Important Factoids
Nothing special that I can think of.

**Comment by angelichorsey** [comment]
I am able to deploy from a template using the same account without any issues. I notice that this error happens during the `virtualmachine.Clone` process.
https://github.com/terraform-providers/terraform-provider-vsphere/blob/1863be0aa50808140a833bde9dfffd5acc142bcb/vsphere/internal/helper/virtualmachine/virtual_machine_helper.go#L203

What's the closest way I can replicate this in the vsphere client GUI? There is no clone to vm option on the template. Only clone to template and convert to vm.

**Comment by angelichorsey** [comment]
Solved: The resource pool I was using was under a different cluster than where I was creating the VM. Decided to manually specify the host ID and recevied a new error saying the host wasn't a part of the resource pool. I also played around in `govc` and `jq` to review all the information about a ManagedObject.

`govc host.info -host.dns=some.domain -json | jq . > host_info.json`

Then I just searched through that file for extra information.
